service/src/main/java/gov/nasa/pds/api/registry/search/RegistrySearchRequestBuilder.java

```diff
@@ -53,222 +52,187 @@ public class RegistrySearchRequestBuilder {
-      } else if (searchAfter.size() != sort.size()) {
+      } else if (searchAfterFieldValues.size() != sortFieldNames.size()) {
         throw new SortSearchAfterMismatchException("sort and searchAfter arguments must be of equal length if provided");
       }
-      this.searchAfter(searchAfter);
+      this.searchAfterFromStrings(searchAfterFieldValues);
     }
 
     return this;
 
   }
 
-  public RegistrySearchRequestBuilder sort(List<String> sort) {
+  /**
+   * Implements an alternative to .sort() that accepts strings in API property format.
+   * Currently hardcoded to sort in ascending order only.
+   * @param sortFieldNames
+   */
+  public RegistrySearchRequestBuilder sortFromStrings(List<String> sortFieldNames) {
 
     String openSearchField;
 
     List<SortOptions> sortOptionsList = new ArrayList<SortOptions>();
-    for (String field : sort) {
+    for (String field : sortFieldNames) {
       openSearchField = SearchUtil.jsonPropertyToOpenProperty(field);
       FieldSort fieldSort =
           new FieldSort.Builder().field(openSearchField).order(SortOrder.Asc).build();
       sortOptionsList.add(new SortOptions.Builder().field(fieldSort).build());
     }
 
-    this.searchRequestBuilder.sort(sortOptionsList);
+    this.sort(sortOptionsList);
 
     return this;
 
   }
 
-  public RegistrySearchRequestBuilder size(Integer size) {
-    this.searchRequestBuilder.size(size);
-    return this;
-  }
-
-  public RegistrySearchRequestBuilder searchAfter(List<String> searchAfter) {
-    // TODO check if the number value need to be handled specfically.
-
-
+  /**
+   * Implements an alternative to .searchAfter() that accepts values as strings.
+   * No-op in current version of OpenSearch client, but a later version will require the commented-out
+   * implementation to convert the Strings to FieldValues
+   * @param searchAfterValues
+   */
+  public RegistrySearchRequestBuilder searchAfterFromStrings(List<String> searchAfterValues) {
     /*
-     * This code is useless with the version of opensearch client we are using From the source code
-     * of the opensearch client in a later version, that is what will need to be used. I am keeping
-     * it here as a comment.
-     * 
      * List<FieldValue> fieldValues = new ArrayList<FieldValue>();
      * 
      * for (String fieldValue : searchAfter) { fieldValues.add(new
+    // TODO check if the number value need to be handled specfically. Method stringValue() implies yes
      * FieldValue.Builder().stringValue(fieldValue).build()); }
      */
-    this.searchRequestBuilder.searchAfter(searchAfter);
+    this.searchAfter(searchAfterValues);
 
 
     return this;
   }
 
 
-  public RegistrySearchRequestBuilder addQParam(String q) throws UnparsableQParamException {
+  /**
+   * Implements an alternative to .fields() that accepts values as strings.
+   * @param fieldNames
+   */
+  public RegistrySearchRequestBuilder fieldsFromStrings(List<String> fieldNames) {
```

Comment:
same as above,but in this case this is confusing because fieldsFromStrings is not a wrapper of fields but of source ! Which is confusing. Not big deal anyway, we'll find an elegant solution.

Reply:
The rationale for FromStrings here is for consistency with the other *FromStrings methods - they're all doing basically the same thing, abstracting away the verbose construction of proper classed objects from Strings

If I were writing it from scratch, I'd have named it `sourceIncludesFromStrings()` instead, since it's setting the _source _includes, as you say, but the method was already there and I assumed it had been named `fields()` because you found it more intuitive.  If not, I'll change it to `sourceIncludesFromStrings()` if you agree that's an improvement?